yt/frontends/ramses/data_structures.py

```diff
@@ -582,6 +583,21 @@ def read_rhs(f, cast):
         if self.num_groups > 0:
             self.group_size = rheader['ncpu'] // self.num_groups
 
+        # Read namelist.txt file (if any)
+        self.read_namelist()
+
+    def read_namelist(self):
+        """Read the namelist.txt file in the output folder, if present"""
+        namelist_file = os.path.join(self.root_folder, 'namelist.txt')
+        if os.path.exists(namelist_file):
+            try:
+                with open(namelist_file, 'r') as f:
+                    nml = f90nml.read(f)
+            except ImportError as e:
+                nml = "An error occurred when reading the namelist: %s" % str(e)
```

Comment:
If we make this an exception of some type, we might be more easily able to determine if it's the success or not, with an `isinstance`.

Reply:
I am not sure I understand your comment.

However, the reason I settled for this is that reading `namelist.nml` is not required to properly load the simulation, so I was trying to avoid throwing errors at the user.

Reply:
NBD, just thought if we had an object we could check if it read correctly without doing a string comparison.  I agree about not throwing errors!

Reply:
I am still confused ; where do you see a string comparison? Here I am storing the content of the namelist in `ds.parameters['namelist']`. If we're missing the `f90nml` package, we store the error message instead.

That is probably not the best solution though.

Reply:
Sorry, I meant -- let's pretend I wanted to know if the namelist had been parsed.  I could do that with an `isinstance` or I could check the contents of the string, right?  Or is it only going to be an instance of `str` if it succeeds?

Reply:
There are three cases:

1. the file does not exist and `ds.parameters['namelist']` does not exist,
2. the file exists and `f90nml` is missing: `ds.parameters['namelist']` then contains the error message as a string,
3. the file exists and `f90nml` is installed: `ds.parameters['namelist']` is a dict-like structure with the content of the namelist.

Any idea to make it better is welcome!

Reply:
So right now if I wanted to see if the namelist was parsed, I wouldn't have a proper way of doing it except checking `isinstance(ds['parameters'], str)`.